packages/datadog-plugin-openai/src/index.js

```diff
@@ -528,18 +540,27 @@ function commonCreateResponseExtraction (tags, body, store) {
 
   store.choices = body.choices
 
-  for (let i = 0; i < body.choices.length; i++) {
-    const choice = body.choices[i]
-    tags[`openai.response.choices.${i}.finish_reason`] = choice.finish_reason
-    tags[`openai.response.choices.${i}.logprobs`] = ('logprobs' in choice) ? 'returned' : undefined
-    tags[`openai.response.choices.${i}.text`] = truncateText(choice.text)
+  for (let choiceIdx = 0; choiceIdx < body.choices.length; choiceIdx++) {
+    const choice = body.choices[choiceIdx]
+    tags[`openai.response.choices.${choiceIdx}.finish_reason`] = choice.finish_reason
+    tags[`openai.response.choices.${choiceIdx}.logprobs`] = ('logprobs' in choice) ? 'returned' : undefined
+    tags[`openai.response.choices.${choiceIdx}.text`] = truncateText(choice.text)
 
     // createChatCompletion only
     if ('message' in choice) {
       const message = choice.message
-      tags[`openai.response.choices.${i}.message.role`] = message.role
-      tags[`openai.response.choices.${i}.message.content`] = truncateText(message.content)
-      tags[`openai.response.choices.${i}.message.name`] = truncateText(message.name)
+      tags[`openai.response.choices.${choiceIdx}.message.role`] = message.role
+      tags[`openai.response.choices.${choiceIdx}.message.content`] = truncateText(message.content)
+      tags[`openai.response.choices.${choiceIdx}.message.name`] = truncateText(message.name)
+      if ('tool_calls' in message) {
```

Comment:
`in` checks should generally be avoided as they can be corrupted by prototype-pollution without a `hasOwnProperty` check. Probably doesn't _really_ matter in this particular case, but it's probably something we should be linting out.

Reply:
sounds good, can change this syntax. had it this way conforming with the style the file was originally written, but will make the change all around!

Reply:
Ah, I probably missed that in the original PR then. 😅